pkg/webhook/resources/virtualmachineimage/mutator.go

```diff
@@ -0,0 +1,116 @@
+package virtualmachineimage
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	longhorntypes "github.com/longhorn/longhorn-manager/types"
+	ctlstoragev1 "github.com/rancher/wrangler/pkg/generated/controllers/storage/v1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/util"
+	"github.com/harvester/harvester/pkg/webhook/types"
+)
+
+func NewMutator(storageClassCache ctlstoragev1.StorageClassCache) types.Mutator {
+	return &virtualMachineImageMutator{
+		storageClassCache: storageClassCache,
+	}
+}
+
+type virtualMachineImageMutator struct {
+	types.DefaultMutator
+	storageClassCache ctlstoragev1.StorageClassCache
+}
+
+func (m *virtualMachineImageMutator) Resource() types.Resource {
+	return types.Resource{
+		Names:      []string{harvesterv1.VirtualMachineImageResourceName},
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   harvesterv1.SchemeGroupVersion.Group,
+		APIVersion: harvesterv1.SchemeGroupVersion.Version,
+		ObjectType: &harvesterv1.VirtualMachineImage{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+		},
+	}
+}
+
+func (m *virtualMachineImageMutator) Create(request *types.Request, newObj runtime.Object) (types.PatchOps, error) {
+	newImage := newObj.(*harvesterv1.VirtualMachineImage)
+
+	return m.patchImageStorageClassParams(newImage)
+}
+
+func (m *virtualMachineImageMutator) patchImageStorageClassParams(newImage *harvesterv1.VirtualMachineImage) ([]string, error) {
+	var patchOps types.PatchOps
+
+	storageClassName := newImage.Annotations[util.AnnotationStorageClassName]
+	storageClass, err := m.getStorageClass(storageClassName)
+	if err != nil {
+		return patchOps, err
```

Comment:
I thought that would return the isNotFound error. 
Could we also add some error messages here to make debugging easy?

Reply:
Any error here should return
already has log message in https://github.com/harvester/harvester/blob/4a972de680447102c78d8831741a423f674448e2/pkg/webhook/types/admission.go#L107

Reply:
Ah, did not trace on the upper level... Thanks for the tips!